R/import_ebird.R

```diff
@@ -1,51 +1,113 @@
+#' of these datasets as tar archive files. `import_ebird()` takes the path to a
+#' tar file as input and imports the text file contained within to a parquet
+#' file, which will allow much easier access to the data. This function will
+#' automatically detect whether you are importing a checklist or observation
+#' dataset provided you **do not change the name of the downloaded file or
+#' unarchive the tar file**. The parquet files will be stored in the directory
+#' specified by [ebird_data_dir()], consult the help for that function to learn
+#' how to modify the parquet directory.
 #' 
+#' @return Invisibly return the path to the directory containing eBird parquet
+#'   files.
 #' @export
 #' @examples
+#' # only use a tempdir for this example, don't copy this for real data
+#' temp_dir <- file.path(tempdir(), "birddb")
+#' Sys.setenv("BIRDDB_HOME" = temp_dir)
 #' 
-#' Sys.setenv("BIRDDB_HOME"=tempdir())
-#' tar <- ebird_sample_data()
+#' # get the path to a sample dataset provided with the package
+#' tar <- sample_observation_data()
+#' # import the sample dataset to parquet
 #' import_ebird(tar)
 #' 
+#' unlink(temp_dir, recursive = TRUE)
 import_ebird <- function(tarfile) {
+  file_metadata <- parse_ebd_filename(tarfile)
+  dest <- file.path(ebird_data_dir(), file_metadata[["dataset"]])
   
+  # extract the tarfile to a temp directory
   source_dir <- tempfile("ebird_tmp")
-  dest <- file.path(ebird_data_dir(), "parquet")
-  
   dir.create(source_dir, recursive = TRUE)
   utils::untar(tarfile = tarfile, exdir = source_dir)
   ebd <- list.files(source_dir, pattern = "ebd.*\\.txt\\.gz",
                    full.names = TRUE, recursive = TRUE)
+  if (length(ebd) != 1 || !file.exists(ebd)) {
+    stop("txt.gz file not successfully extracted from tarfile.")
+  }
   
+  # open tsv and stream to parquet
   ds <- arrow_open_ebird_txt(ebd, dest)
   
-  # clean up column names
-  col_names <- names(ds)
-  names(col_names) <- gsub("[/ ]", "_", tolower(col_names))
-  ds <- dplyr::select(ds, dplyr::all_of(col_names))
-  
-  # Consider alternative partitions that might speed common queries
-  # partitioning = c("COUNTRY")
-  # However, arrow supports at most 1024 partitions
+  # confirm overwrite
+  if (dir.exists(dest)) {
+    if (interactive()) {
+      msg <- paste("eBird", file_metadata[["dataset"]], 
+                   "data already exists in BIRDDB_HOME,",
+                   "would you like to overwrite this data?")
+      overwrite <- utils::askYesNo(msg, default = NA)
+      if (isTRUE(overwrite)) {
+        unlink(dest, recursive = TRUE)
+      } else {
+        stop("Cancelling data import to avoid overwriting existing data.")
+      }
+    } else {
+      message("Overwriting existing eBird ", 
+              file_metadata[["dataset"]], 
+              " data.")
+      unlink(dest, recursive = TRUE)
+    }
+  }
   
+  # stream to parquet
   arrow::write_dataset(ds, dest, format = "parquet")
   
+  # save metadata
```

Comment:
as per comment above, seems like 'save_metadata() could be a one-line call to a `record_metadata()` fn instead of 4 lines.  seems to me like a clearer distinction between the roles of each fn here, but obviously that's me being pedantic.